backend/associations/serializers/library.py

```diff
@@ -137,13 +137,16 @@ def get_user_loan(self, loanable):
 
         return None
 
-    def get_number_of_pending(self, loanable):
-        """Return the number of pending requests for this loanable."""
-        return len(loanable.loans.filter(status="PENDING"))
+    def get_number_of_pending_loans(self, loanable):
+        """Return the number of pending loans for this loanable."""
+        return loanable.number_of_pending_loans
+
+    def to_representation(self, loanable: Loanable):
+        res = super().to_representation(loanable)
+
+        res["status"] = "AVAILABLE" if loanable.is_available() else "BORROWED"
+        res["expected_return_date"] = loanable.get_expected_return_date()
```

Comment:
Not sure I understood your remark, but you want to know, at the loanable level, when it will be returned, without having to dig into the loans

Reply:
Yes, but here we're looking at `Loan`. Let's say you  want to know it a loanable was given back late. You'll look at `expected_return_date` and `actual_return_date`, but since `expected_return_date` is not the Loan one but the Loanable one, you'll never be able to get that information

Reply:
> Yes, but here we're looking at Loan.

This is a `Loanable` serializer. I don't understand your point

> Let's say you want to know it a loanable was given back late. You'll look at expected_return_date and actual_return_date, but since expected_return_date is not the Loan one but the Loanable one, you'll never be able to get that information

Yes in that case you look at the `Loan`, not at the `Loanable`.